flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/internal/ShowCreateUtil.java

```diff
@@ -0,0 +1,221 @@
+                                                            watermarkSpec.getRowtimeAttribute()),
+                                                    watermarkSpec
+                                                            .getWatermarkExpression()
+                                                            .asSerializableString()))
+                            .collect(Collectors.joining("\n")));
+        }
+        return Optional.empty();
+    }
+
+    static Optional<String> extractFormattedComment(ResolvedCatalogBaseTable<?> table) {
+        String comment = table.getComment();
+        if (StringUtils.isNotEmpty(comment)) {
+            return Optional.of(EncodingUtils.escapeSingleQuotes(comment));
+        }
+        return Optional.empty();
+    }
+
+    static Optional<String> extractFormattedPartitionedInfo(ResolvedCatalogTable catalogTable) {
+        if (!catalogTable.isPartitioned()) {
+            return Optional.empty();
+        }
+        return Optional.of(
+                catalogTable.getPartitionKeys().stream()
+                        .map(EncodingUtils::escapeIdentifier)
+                        .collect(Collectors.joining(", ")));
+    }
+
+    static Optional<String> extractFormattedOptions(
+            ResolvedCatalogBaseTable<?> table, String printIndent) {
+        if (Objects.isNull(table.getOptions()) || table.getOptions().isEmpty()) {
+            return Optional.empty();
+        }
+        return Optional.of(
+                table.getOptions().entrySet().stream()
+                        .map(
+                                entry ->
+                                        String.format(
+                                                "%s\'%s\' = \'%s\'",
+                                                printIndent, entry.getKey(), entry.getValue()))
+                        .collect(Collectors.joining(",\n")));
+    }
+
+    static String extractFormattedColumnNames(ResolvedCatalogBaseTable<?> baseTable) {
+        return baseTable.getResolvedSchema().getColumns().stream()
+                .map(Column::getName)
+                .map(EncodingUtils::escapeIdentifier)
+                .collect(Collectors.joining(", "));
+    }
+
+    static String buildShowCreateTableRow(
+            ResolvedCatalogBaseTable<?> table,
+            ObjectIdentifier tableIdentifier,
+            boolean isTemporary) {
+        if (table.getTableKind() == CatalogBaseTable.TableKind.VIEW) {
+            throw new TableException(
+                    String.format(
+                            "SHOW CREATE TABLE is only supported for tables, but %s is a view. Please use SHOW CREATE VIEW instead.",
+                            tableIdentifier.asSerializableString()));
+        }
+        final String printIndent = "  ";
+        StringBuilder sb =
+                new StringBuilder()
+                        .append(buildCreateFormattedPrefix("TABLE", isTemporary, tableIdentifier));
+        // append columns
```

Comment:
nit: all of these comments are redundant and noisy now because the method names already make it clear what's happening